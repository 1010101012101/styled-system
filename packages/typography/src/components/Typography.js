import React from 'react'
import styled from '@emotion/styled'
import { Global } from '@emotion/core'
import {
  compose,
  fontFamily,
  fontSize,
  fontWeight,
  lineHeight,
  space,
  color,
  maxWidth,
} from 'styled-system'
import pick from 'lodash.pick'
import {
  elementStyles,
  typography,
  defaultStyles
} from './element-styles'

const rootTypography = ({ theme, root }) => typography({ theme, ...root })
const Root = styled.div(
  defaultStyles,
  rootTypography,
  elementStyles,
  // handle other/nested CSS
  props => props.css
)

// may not need the wrapper
export const Typography = ({ html, ...props }) =>
  <>
    {html && <Global styles={{ html }} />}
    <Root {...props} />
  </>

export default Typography
