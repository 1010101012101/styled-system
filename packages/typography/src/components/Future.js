// Future Typography Theme
import React from 'react'
import Typography from './index'
import merge from 'lodash.merge'

const config = {
  color: 'text',
  bg: 'background',
  fontFamily: '"Avenir Next", system-ui, sans-serif',
  lineHeight: 1.5,
  px: [3, 4],
  py: 4,
  a: {
    color: 'link',
  },
  h1: {
    fontSize: [5, 6],
    fontWeight: 600,
    my: 5,
    textTransform: 'uppercase',
    letterSpacing: '0.2em',
  },
  pre: {
    fontFamily: 'monospace',
    color: 'highlight',
    bg: 'black',
    p: 3,
    borderRadius: 8,
    overflowX: 'auto',
  },
  code: {
    fontFamily: 'monospace',
    fontSize: '85%',
    color: 'link',
    bg: 'black',
  },
  p: {
    fontSize: [2, 3],
    lineHeight: 1.625,
    my: 4,
    maxWidth: '40em',
  },
  css: {
    'pre code': {
      fontSize: 'inherit',
      color: 'inherit',
    },
    'li > p': {
      fontSize: 'inherit',
      margin: 0,
    }
  }
}

export const Future = props =>
  <Typography
    {...merge(config, props)}
  />

Future.defaultProps = {
  theme: {
    fonts: {
      monospace: 'Menlo, monospace',
    },
    colors: {
      text: '#00ccff',
      background: '#000e12',
      highlight: '#ff4ca5',
      black: '#000',
      link: '#99ebff',
    }
  }
}

export default Future
