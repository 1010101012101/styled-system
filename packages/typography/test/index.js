import React from 'react'
import TestRenderer from 'react-test-renderer'
import serializer from 'jest-emotion'
import Typography from '../src'
import Future from '../src/future'

expect.addSnapshotSerializer(serializer)

const renderJSON = el => TestRenderer.create(el).toJSON()

test('renders', () => {
  const json = renderJSON(<Typography />)
  expect(json).toMatchSnapshot()
})

test('renders with props', () => {
  const json = renderJSON(
    <Typography
      fontFamily='fantasy'
      lineHeight='2'
      color='tomato'
    />
  )
  expect(json).toMatchSnapshot()
})

test('renders with element props', () => {
  const json = renderJSON(
    <Typography
      h1={{
        fontSize: 3,
        fontWeight: 900,
      }}
    />
  )
  expect(json).toMatchSnapshot()
})

// seems to fail serializer
test.skip('renders with responsive element props', () => {
  const json = renderJSON(
    <Typography
      h1={{
        fontSize: [3, 4, 5],
        fontWeight: 900,
        my: [3, 4],
        color: 'tomato'
      }}
    />
  )
  expect(json).toMatchSnapshot()
})

test.skip('Future theme renders', () => {
  const json = renderJSON(<Future />)
  expect(json).toMatchSnapshot()
})
