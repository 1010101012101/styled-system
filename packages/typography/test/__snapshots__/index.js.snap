// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
.emotion-0 img {
  max-width: 100%;
  height: auto;
}

<div
  className="emotion-0"
/>
`;

exports[`renders with element props 1`] = `
.emotion-0 img {
  max-width: 100%;
  height: auto;
}

.emotion-0 h1 {
  font-size: 3px;
  font-weight: 900;
  font-size: 20px;
  font-weight: 900;
}

<div
  className="emotion-0"
/>
`;

exports[`renders with props 1`] = `
.emotion-0 {
  font-family: fantasy;
  line-height: 2;
  color: tomato;
}

.emotion-0 img {
  max-width: 100%;
  height: auto;
}

<div
  className="emotion-0"
  color="tomato"
  fontFamily="fantasy"
/>
`;

exports[`renders with responsive element props 1`] = `
.emotion-0 img {
  max-width: 100%;
  height: auto;
}

.emotion-0 h1 {
  font-size: 20px;
  font-weight: 900;
  mt: 3px;
  color: tomato;
  margin-top: 16px;
}

@media screen and (min-width:40em) {
  .emotion-0 h1 {
    font-size: 24px;
  }
}

@media screen and (min-width:52em) {
  .emotion-0 h1 {
    font-size: 32px;
  }
}

<div
  className="emotion-0"
/>
`;
